src/towncrier/_settings/load.py

```diff
@@ -85,6 +85,26 @@ def load_config_from_options(
     return base_directory, config
 
 
+def traverse_for_config(path: str | None) -> tuple[str, Config]:
+    """
+    Search for a configuration file in the current directory and all parent directories.
+
+    Returns the directory containing the configuration file and the parsed configuration.
+    """
+    start_directory = directory = os.path.abspath(path or os.getcwd())
+    while True:
+        config = load_config(directory)
+        if config is not None:
+            return directory, config
+
+        parent = os.path.dirname(directory)
+        if parent == directory:
+            raise ConfigError(
```

Comment:
Do we have a test for this case ?

I don't see any report for missing coverate, but to me it looks like new code, without new tests :)

Reply:
Covered by existing `test_needs_config` test which used:
```python
        self.assertTrue(result.output.startswith("No configuration file found."))
```